### ISSUE TITLE
XLPyDLLActivateAuto(Py, XLPyCommand, 1) raises error

### ISSUE THREAD

**Issue by bjrnfrdnnd** [ZoomerAnalytics/xlwings#764]
Hi there, 

I have a weird problem.
xlwings worked nicely 6 hours ago.
It now stopped working with this error:
![image](https://user-images.githubusercontent.com/5217684/31313327-45c1a2f4-abe0-11e7-99b1-eb9b5cb65f11.png)

This happens when I 
a) try to update import udf functions using the ribbon
b) try to run udf functions which worked before

I am running this on an example created with xlwings quickstart dfdf.
Nothing added in addition, just the plain example.

I am running xlwings 0.11.4 on python 3.6.2 using anaconda 1.6.8 64 bit (installer is called Anaconda3-5.0.0-Windows-x86_64.exe).
This is excel 2013 32 bit running on a lenovo thinkpad with windows 10 enterprise 64 bit,
The processor is i5-7300U. 

When I run the import udf function via the ribbon, the program goes through the xlwings.xlam Main module where it identifies itself as Win32 in the declaration part:
![image](https://user-images.githubusercontent.com/5217684/31313629-875afa82-abe8-11e7-835f-d44dc4906dff.png)

I think this has something to do with 32 vs 64 bit things, but I am not sure what. 
I reinstalled anaconda multiple times, restarted, still the same error.
References to xlwings are set (xlwings.xlam is in the XLSTART folder, the excel options section references it:
![image](https://user-images.githubusercontent.com/5217684/31313397-6c7fee8a-abe2-11e7-9be7-28f05e8e67c0.png)

The add-ins have a reference to it:
![image](https://user-images.githubusercontent.com/5217684/31313405-ad4fbe18-abe2-11e7-939c-68f06b96b046.png)

And I trust access to the vba project object model:
![image](https://user-images.githubusercontent.com/5217684/31313406-c79b16dc-abe2-11e7-97ad-ef269723b865.png)

The vba code has a reference to xlwings.xlam:

![image](https://user-images.githubusercontent.com/5217684/31313414-f52aa8ba-abe2-11e7-9f2c-c137f4c4e0e3.png)

I think I did everything as in the instructions (and it worked a few hours ago).
But now I just cannot get it to work anymore - and I definitely need UDFs.

I specified complete paths to the python executable in the tab xlwings.conf because I do not have administrative access, and the terminal window does not find a python executable:
![image](https://user-images.githubusercontent.com/5217684/31313290-f1865492-abde-11e7-990d-834f33e7ebce.png)

I tried doubling the backslashes, removing the entry for pythonpath, using python or pythonw with and without .exe. All to no avail.

I always get the same error.

which fails here:
![image](https://user-images.githubusercontent.com/5217684/31313355-35043c6e-abe1-11e7-85c5-d56e11f07a91.png)

The XLPyCommand is
``
C:\local\Anaconda3\python -B -c "import sys, os;sys.path.extend(os.path.normcase(os.path.expandvars(r'C:\Users\USERNAME\FV tool\dfdf;C:\local\Anaconda3')).split(';'));import xlwings.server; xlwings.server.serve('$(CLSID)')"
``
When executed in a terminal, this code fails with:
`
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\local\Anaconda3\lib\site-packages\xlwings\server.py", line 291, in serve
    clsid = pywintypes.IID(clsid)
pywintypes.com_error: (-2147221005, 'Invalid class string', None, None)
`

By the way, running the python program from the command line (anaconda prompt) with a custom udf that is being called from main works just fine.
In this case, excel-windows are opened and the result is correctly calculated.
However, running it from excel raises the error.

Just to make that clear: the VBA code identifies itself as Win32, not Win64. XLPyDLLName is xlwings32.dll. This file exists and is located in the same path as the python executable.

Many thanks in advance!

**Comment by bjrnfrdnnd** [comment]
By the way, running the python program from the command line (anaconda prompt) with a custom udf that is being called from __main__ works just fine.
In this case, excel-windows are opened and the result is correctly calculated.
However, running it from excel raises the error.

**Comment by bjrnfrdnnd** [comment]
Just to make that clear: the VBA code identifies itself as Win32, not Win64. I erroneously colored the wrong region blue in one of the above image. XLPyDLLName is xlwings32.dll. This file exists and is located in the same path as the python executable.

**Comment by fzumstein** [comment]
xlwings32.dll corresponds to the version of Excel, so seems all good in your case. I assume you have already rebooted your machine and still goto the same error? Try to run your UDFs in debug mode and see if that works: http://docs.xlwings.org/en/stable/debugging.html#udf-debug-server

**Comment by bjrnfrdnnd** [comment]
Sorry to bother you again.
The whole thing worked under Anaconda python32 in a local (no admin) installation for a week.
Now it stopped working with the same error as above. 
I uninstalled my Anaconda pythons (32 and 64 bit) and reinstalled only the python32 version. No change.
This time, the problem is even weirder: I can run in debug mode and from the commandline (or pycharm) - no problem, everything works. 
The only thing which does not work - and the only thing of importance for a production module - is when I switch from debug mode to no-debug mode. Then, when I hit the "import functions" button of the xlwings ribbon, the error message pops up. When not hitting this button, the error message displays in a cell when running a udf.
Problem now is that I cannot even debug the error, because in debug mode, there are no problems...
So what can I do?

**Comment by fzumstein** [comment]
@bjrnfrdnnd did you ever find out what the error was?

**Comment by MDDietz1** [comment]
Similar problem for me. Everything used to work previously and now fails when running XLPyCommand. If I run it (looks similar  to above) in the PowerShell then I get the following error message: 

```
CLSID : The term 'CLSID' is not recognized as the name of a cmdlet, function, script file, or operable program. Check
the spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:333
+ ... erver.serve('$(CLSID)')"
+                    ~~~~~
    + CategoryInfo          : ObjectNotFound: (CLSID:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException
```

And if I run the commands directly in Python then I get

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\MD123\AppData\Local\Continuum\anaconda3\lib\site-packages\xlwings\server.py", line 301, in serve
    clsid = pywintypes.IID(clsid)
pywintypes.com_error: (-2147221005, 'Invalid class string', None, None)
```
as above.

**Comment by fzumstein** [comment]
@MDDietz1 does it work if you run it in debug mode?

**Comment by MDDietz1** [comment]
No, I am getting

`No command specified in the configuration, cannot autostart server`

And in case that helps:  When I try and run in non-debug mode, I can actually see an instance of pythonw being created in the Task manager, it stays there for 2secs and disappears. That doesn't seem to happen in debug mode (or it is too quick for me  to notice).

**Comment by fzumstein** [comment]
For debug mode, it's not enough to check the box in the ribbon. you will also need to run the python side manually, see here: http://docs.xlwings.org/en/stable/debugging.html#udf-debug-server

**Comment by MDDietz1** [comment]
OK, sorry about that. Behaviour is as follows: 
Python terminal window starts up as expected, showing "xlwings server running, clsid={506.... }"
Code  breaks when executing 
`XLPyDLLActivateAuto(Py, XLPyCommand, 1)`
where 
`XLPyCommand = "{506e67c3-55b5-48c3-a035-eed5deea7d6d}"`

**Comment by MDDietz1** [comment]
OK,  I have now "fixed" the error, by downgrading Python from 3.6.8 to 3.6.7 and then upgrading it again to 3.6.8.  Whatever this did, it fixed the problem.

**Comment by fzumstein** [comment]
thanks for the update!
